erts/emulator/beam/ops.tab

```diff
@@ -1587,6 +1587,19 @@ bif1 Fail u$bif:erlang:float/1 s d => too_old_compiler
 bif1 Fail u$bif:erlang:round/1 s d => too_old_compiler
 bif1 Fail u$bif:erlang:trunc/1 s d => too_old_compiler
 
+#
+# Handle the length/1 guard BIF specially to make it trappable.
+#
+
+gc_bif1 Fail=j Live u$bif:erlang:length/1 Src Dst => \
+   i_length_setup Live Src | i_length Fail Live Dst
+
+i_length_setup t x
```

Comment:
> Would using i_length_setup t xyc work in here?

Yes, it would. Fixed.

> Additionally, given i_length_setup is so tiny - would it make sense to also define it for r? It shouldn't grow the main loop significantly, but might save some space in the bytecode.

That would not save any BEAM code space, as the `t` and `x` operands are packed into the same 32-bit word. It could possibly save a tiny, tiny amount of execution time, but not enough to make it worthwhile.

While looking at the generated code, I realized that I had forgotten to implement a minor optimization that I had thought of earlier.